parteh/PRTAllometricCNPMod.F90

```diff
@@ -0,0 +1,2254 @@
+
+   subroutine TargetAllometryCheck(bleaf,bfroot,bsap,bstore,bdead, &
+                                   bt_leaf,bt_froot,bt_sap,bt_store,bt_dead, &
+                                   grow_leaf,grow_froot,grow_sapw,grow_store)
+
+     ! Arguments
+     real(r8),intent(in) :: bleaf   !actual
+     real(r8),intent(in) :: bfroot
+     real(r8),intent(in) :: bsap
+     real(r8),intent(in) :: bstore
+     real(r8),intent(in) :: bdead
+     real(r8),intent(in) :: bt_leaf   !target
+     real(r8),intent(in) :: bt_froot
+     real(r8),intent(in) :: bt_sap
+     real(r8),intent(in) :: bt_store
+     real(r8),intent(in) :: bt_dead
+     logical,intent(out) :: grow_leaf  !growth flag
+     logical,intent(out) :: grow_froot
+     logical,intent(out) :: grow_sapw
+     logical,intent(out) :: grow_store
+
+     if( (bt_leaf - bleaf)>calloc_abs_error) then
+        write(fates_log(),*) 'leaves are not on-allometry at the growth step'
+        write(fates_log(),*) 'exiting',bleaf,bt_leaf
+        call endrun(msg=errMsg(sourcefile, __LINE__))
+     elseif( (bleaf - bt_leaf)>calloc_abs_error) then
+        ! leaf is above allometry, ignore
+        grow_leaf = .false.
+     else
+        grow_leaf = .true.
+     end if
+
+     if( (bt_froot - bfroot)>calloc_abs_error) then
+        write(fates_log(),*) 'fineroots are not on-allometry at the growth step'
+        write(fates_log(),*) 'exiting',bfroot, bt_froot
+        call endrun(msg=errMsg(sourcefile, __LINE__))
+     elseif( ( bfroot-bt_froot)>calloc_abs_error ) then
+        grow_froot = .false.
+     else
+        grow_froot = .true.
+     end if
+
+     if( (bt_sap - bsap)>calloc_abs_error) then
+        write(fates_log(),*) 'sapwood is not on-allometry at the growth step'
+        write(fates_log(),*) 'exiting',bsap, bt_sap
+        call endrun(msg=errMsg(sourcefile, __LINE__))
+     elseif( ( bsap-bt_sap)>calloc_abs_error ) then
+        grow_sapw = .false.
+     else
+        grow_sapw = .true.
+     end if
+
+     if( (bt_store - bstore)>calloc_abs_error) then
+        write(fates_log(),*) 'storage is not on-allometry at the growth step'
+        write(fates_log(),*) 'exiting',bstore,bt_store
+        call endrun(msg=errMsg(sourcefile, __LINE__))
+     elseif( ( bstore-bt_store)>calloc_abs_error ) then
+        grow_store = .false.
+     else
+        grow_store = .true.
+     end if
+
+     if( (bt_dead - bdead)>calloc_abs_error) then
+        write(fates_log(),*) 'structure not on-allometry at the growth step'
+        write(fates_log(),*) 'exiting',bdead,bt_dead
+        call endrun(msg=errMsg(sourcefile, __LINE__))
+     end if
+   end subroutine TargetAllometryCheck
+
+   ! =====================================================================================
+
+   subroutine FastPRTAC(this)
+
+      implicit none
+      class(cnp_allom_prt_vartypes) :: this     ! this class
+
+      ! This routine does nothing, because in the carbon only allometric RT model
```

Comment:
CNP?

Reply:
removed this routine, it was obsolete anyway